packages/common/http/test/headers_spec.ts

```diff
@@ -46,6 +46,20 @@ import {HttpHeaders} from '@angular/common/http/src/headers';
 
         expect(headers.getAll('foo')).toEqual(['second']);
       });
+
+      it('should throw an error when null is passed as header', () => {
+        const headers = new HttpHeaders({foo: null!});
```

Comment:
I'd propose adding a comment here to explain that the `strictNullChecks` may affect the typing and allow null/undefined values:
```suggestion
        // Note: the `strictNullChecks` set to `false` in TS config would make `null` and `undefined`
        // valid values within the headers object, thus this test verifies this scenario.
        const headers = new HttpHeaders({foo: null!});
```
The second test would benefit from the same comment as well.